packages/next/client/components/layout-router.tsx

```diff
@@ -101,11 +102,18 @@ function findDOMNode(
 }
 
 /**
- * Check if the top of the HTMLElement is in the viewport.
+ * Check if the top-left corner of the HTMLElement is in the viewport.
  */
-function topOfElementInViewport(element: HTMLElement) {
+function topLeftOfElementInViewport(element: HTMLElement) {
```

Comment:
Might have to check for right-to-left languages in the future.

Reply:
:+1: haven't thought about that.

Reply:
Ok, after a little thought that is not worth it. I'm removing horizontal scrolling since it will make the tests and code complicated for very little benefit.

Reply:
Added comment explaining the current situation and removed horzontal scrolling